### PR TITLE
Move order walkthrough to frontend subproject

### PR DESCRIPTION
**Pull request by mbj**
Extraction from #398

### CODE REVIEW

**Comment by mbj** [comment]
@dkubb This is ready for review.

File: frontend/spec/support/order_walkthrough.rb [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+    STATES = IceNine.deep_freeze(%i[address delivery payment complete])
+
+    private_constant(*constants(false))
+
+    def self.up_to(state)
+      # Need to create a valid zone too...
+      zone    = SpreeSpec::Zone.global
+      country = FactoryGirl.create(:country)
+
+      zone.members << Spree::ZoneMember.create(:zoneable => country)
```
Comment (dkubb) [inline frontend/spec/support/order_walkthrough.rb]: Use 1.9 style hashes.

File: frontend/spec/support/order_walkthrough.rb [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+    end
+
+    def delivery(order)
+      order.next!
+    end
+
+    def payment(order)
+      order.payments.create!(
+        payment_method: Spree::PaymentMethod.first!,
+        amount: order.total
```
Comment (dkubb) [inline frontend/spec/support/order_walkthrough.rb]: Values should be aligned.

File: frontend/spec/support/order_walkthrough.rb [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+
+      order
+    end
+
+    private
+
+    def address(order)
+      state = country.states.first!
+
+      order.bill_address = FactoryGirl.create(
```
Comment (dkubb) [inline frontend/spec/support/order_walkthrough.rb]: Given these two blocks are the same you could simplify it by doing:

```ruby
%i[bill_address ship_address].each do |association|
  order[association] = FactoryGirl.create(
    :address,
    state:   state,
    country: country
  )
end
```

**Comment by dkubb** [comment]
@mbj I finished a pass over this PR.

**Pull request merged**